runtime/lua/vim/treesitter/query.lua

```diff
@@ -114,9 +114,12 @@ local explicit_queries = setmetatable({}, {
 ---
 ---@param lang string: The language to use for the query
 ---@param query_name string: The name of the query (i.e. "highlights")
----@param text string: The query text (unparsed).
-function M.set_query(lang, query_name, text)
-  explicit_queries[lang][query_name] = M.parse_query(lang, text)
+---@param query string or Query: The query text (either unparsed or parsed).
+function M.set_query(lang, query_name, query)
```

Comment:
Isn't this API call always bad and should be deprecated? Ideally you want always to be able to set unparsed queries and having them parsed only once they are needed.

Reply:
I assume you are using the `set_query` from your config which is loaded on startup. Parsing the queries takes quite some time while you only want to set a config value and should only be done when the filetype is loaded just like it is done with the default config.

https://github.com/neovim/neovim/pull/14932

Reply:
How about the `explicit_queries` actually store the unparsed queries and they are only parsed before being returned from `get_query`? Something like:
```lua
function M.set_query(lang, query_name, text)
  -- text is a string
  explicit_queries[lang][query_name] = text
end
function M.get_query(lang, query_name)
  if explicit_queries[lang][query_name] then
    return M.parse_query(lang, explicit_queries[lang][query_name])
  end
  ...
end
```
or should it be delayed even further outside of `query.lua`?

If so then maybe there could also be a `get_raw_query` which doesn't parse the query before returning which would be useful in my case, because then the return of `get_raw_query` can be passed to `set_query`.

> I assume you are using the set_query from your config which is loaded on startup.

Actually not in my case, I have it in a opt-in toggle function to temporarily inject some custom language in the current language strings. Anyway, my specific usecase is maybe not the most relevant, not sure :)

Reply:
@theHamsta Any more thoughts on this?

(btw, let me know if reminders like this is discouraged, just not sure how the workflow is)